src/gui/attributetable/qgsattributetablemodel.cpp

```diff
@@ -698,17 +699,29 @@ QVariant QgsAttributeTableModel::data( const QModelIndex &index, int role ) cons
   }
 
   QVariant val = mFeat.attribute( fieldId );
+  bool isLink = val.type() == QVariant::String && QgsStringUtils::isUrl( val.toString() );
```

Comment:
Again, given that we'll be evaluating this many millions of times, can you only evaluate this when the role is one which requires this knowledge? (ie tooltip/foreground role/font role)